packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -747,8 +756,17 @@ def _serialize_body_parameter(self, builder: OperationType) -> List[str]:
 
         This function serializes the body params that need to be serialized.
         """
-        retval: List[str] = []
         body_param = cast(BodyParameter, builder.parameters.body_parameter)
+        if body_param.default_content_type == "multipart/form-data":
+            return [
+                f"if isinstance({body_param.client_name}, _model_base.Model):",
+                f"    _body = {body_param.client_name}.as_origin_dict()",
+                "else:",
+                f"    _body = cast(Dict[str, Any], {body_param.client_name})",
```

Comment:
i don't like having this `cast` in generated code. What is this trying to solve?

Reply:
`as_origin_dict` return type is `Dict[str, Any] while SDK body type is `JSON`, then mypy report incompatible type error without cast

Reply:
fixed